src/Symfony/Component/AstGenerator/AstGeneratorChain.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\AstGenerator;
+
+/**
+ * Generator delegating the generation to a chain of generators.
+ *
+ * @author Joel Wurtz <joel.wurtz@gmail.com>
+ */
+class AstGeneratorChain implements AstGeneratorInterface
```

Comment:
What do you think about talking about that later in another pr ? :) This is already a complex topic and i would like to keep this pr as small as possible (so maybe no extension point at first).